centrallix/multiquery/multiquery.c

```diff
@@ -2120,7 +2123,8 @@ mq_internal_SyntaxParse(pLxSession lxs, pQueryStatement stmt, int allow_empty, p
 					}
 				    }
 
-				strtcpy(new_qs->Source, xs->String, sizeof(new_qs->Source));
+				if (xs != NULL)
+				    strtcpy(new_qs->Source, xs->String, sizeof(new_qs->Source));
```

Comment:
The next_state assignment below should also be within this if statement, otherwise the ParseError state gets overwritten.